### PR TITLE
Stop transfer of ownership between same users

### LINKED ISSUE #23280: Transfer files from<->to same user results in data loss
### Steps to reproduce
1. run `sudo -u www-data php occ files:transfer <source-user> <source-user>` (there is no mistake, both users are the same)

### Expected behaviour
There should be error or warning, risk of recursion and loosing data. 

### Actual behaviour
All user's files were deleted

### PR DESCRIPTION
**Pull request by sharidas**
Fix #23280

Stop the transfer of ownership between same user.

Signed-off-by: Sujith Haridasan <sujith.h@gmail.com>

### CODE REVIEW

**Comment by sharidas** [comment]
### Summary

Transferring ownership between same users look risky.  

### Description

Imho, transferring files/folders of same user looks little risky, hence made this change set. Hope this looks ok.

### How was it tested?

1. Create user `admin`
2. Execute 
```
➜  nextcloud2 git:(master) ✗ ./occ files:transfer-ownership admin admin
PHP Warning:  Module 'pdo_mysql' already loaded in Unknown on line 0
Kindly check the source and destination users, as they are same.
➜  nextcloud2 git:(master) ✗
```

File: apps/files/lib/Command/TransferOwnership.php [hunk 1]
```diff
@@ -87,6 +87,16 @@ protected function configure() {
 	}
 
 	protected function execute(InputInterface $input, OutputInterface $output): int {
+
+		/**
+		 * Check if source and destination users are same. If they are same then just ignore the transfer.
+		 */
+
+		if ($input->getArgument(('source-user')) === $input->getArgument('destination-user')) {
+			$output->writeln("<error>Kindly check the source and destination users, as they are same.</error>");
```
Comment (nickvergessen) [inline apps/files/lib/Command/TransferOwnership.php:96]: :heart:

**Review by nickvergessen — APPROVED** [review]

File: apps/files/lib/Command/TransferOwnership.php (hunk 1 above)
Comment (acsfer) [reply apps/files/lib/Command/TransferOwnership.php:96]: What about a more explicit message like "Ownership can't be transferred when Source and Destination users are the same user. Please check your input."

Comment (sharidas) [reply apps/files/lib/Command/TransferOwnership.php:96]: @acsfer Sure, it sounds better than my message :) I will update it accordingly.

Comment (sharidas) [reply apps/files/lib/Command/TransferOwnership.php]: Updated the message.

**Review by solracsf — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Ci failure is fixed in #23391

**Pull request merged**